test/core/data/datasources/account_remote_data_source_test.dart

```diff
@@ -0,0 +1,152 @@
+import 'package:coffeecard/core/data/datasources/account_remote_data_source.dart';
+import 'package:coffeecard/core/network/network_request_executor.dart';
+import 'package:coffeecard/features/occupation/data/models/occupation_model.dart';
+import 'package:coffeecard/features/user/data/models/user_model.dart';
+import 'package:coffeecard/features/user/domain/entities/role.dart';
+import 'package:coffeecard/generated/api/coffeecard_api.swagger.dart' as v1;
+import 'package:coffeecard/generated/api/coffeecard_api_v2.swagger.dart' as v2;
+import 'package:coffeecard/models/account/authenticated_user.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:fpdart/fpdart.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+
+import 'account_remote_data_source_test.mocks.dart';
+
+@GenerateMocks([v1.CoffeecardApi, v2.CoffeecardApiV2, NetworkRequestExecutor])
+void main() {
+  late MockCoffeecardApi apiV1;
+  late MockCoffeecardApiV2 apiV2;
+  late MockNetworkRequestExecutor executor;
+  late AccountRemoteDataSource repository;
+
+  setUp(() {
+    apiV1 = MockCoffeecardApi();
+    apiV2 = MockCoffeecardApiV2();
+
+    executor = MockNetworkRequestExecutor();
+    repository = AccountRemoteDataSource(
+      apiV1: apiV1,
+      apiV2: apiV2,
+      executor: executor,
+    );
+  });
+
+  group('register', () {
+    test('should call executor', () async {
+      // arrange
+      when(executor.call<v2.MessageResponseDto>(any)).thenAnswer(
+        (_) async => Right(v2.MessageResponseDto()),
+      );
+
+      // act
+      await repository.register(
+        'name',
+        'email',
+        'passcode',
+        0,
+      );
+
+      // assert
+      verify(executor.call(any));
+    });
+  });
+
+  group('login', () {
+    test('should call executor', () async {
```

Comment:
update test name to reflect what's going on (should return authenticated user when executor returns a token...)

Reply:
```suggestion
    test('should return authenticated user when executor returns a token', () async {
```